src/MerchantFrontend/src/app/views/merchant-details/merchant-details.component.ts

```diff
@@ -7,9 +11,20 @@ import { Component, OnInit } from '@angular/core';
 })
 export class MerchantDetailsComponent implements OnInit {
 
-  constructor() { }
+  merchant: Merchant;
+
+  constructor(
+    private loginService: LoginService,
+    private stateService: StateService,
+    private router: Router
+  ) { }
 
   ngOnInit(): void {
+    this.merchant = this.stateService.merchant;
   }
 
+  logout(): void {
+    this.loginService.logout();
+    this.router.navigate(['login', 'merchant']);
```

Comment:
couldn't you redirect to login screen in the loginService?

Reply:
i cannot have the router in the injectiontree of the app initializer, so this was the easier implementation